robot/src/main/java/us/ilite/robot/hardware/NeoDriveHardware.java

```diff
@@ -0,0 +1,330 @@
+            sparkMax.setIdleMode(pNeutralMode);
+        }
+    }
+
+    private void configureMaster(CANSparkMax sparkMax, boolean pIsLeft) {
+        // Velocity, temperature, voltage, and current according the REV docs
+        sparkMax.setPeriodicFramePeriod(CANSparkMaxLowLevel.PeriodicFrame.kStatus1, 5);
+        // Position according to REV docs
+        sparkMax.setPeriodicFramePeriod(CANSparkMaxLowLevel.PeriodicFrame.kStatus2, 5);
+
+        sparkMax.setSmartCurrentLimit(SystemSettings.kDriveCurrentLimitAmps);
+        sparkMax.setSecondaryCurrentLimit(SystemSettings.kDriveCurrentLimitAmps);
+        // Set a peak current limit duration??
+    }
+
+    private void configureMotor(CANSparkMax motorController) {
+        /*
+        TODO Disabled voltage comp for now because of:
+        https://www.chiefdelphi.com/t/sparkmax-voltage-compensation/350540/5
+         */
+//        motorController.enableVoltageCompensation(12.0);
+        // No velocity measurement filter
+        motorController.setOpenLoopRampRate(SystemSettings.kDriveTopOpenLoopVoltageRampRate);
+        motorController.setClosedLoopRampRate(SystemSettings.kDriveClosedLoopVoltageRampRate);
+        // motorController.configNeutralDeadband(0.04, 0);
+    }
+
+    private void configSparkForPercentOutput(CANSparkMax pSparkMax) {
+        // talon.configNeutralDeadband(0.04, 0);
+    }
+
+    private void configSparkForVelocity(CANSparkMax pSparkMax) {
+        mPidSlot = SystemSettings.kDriveVelocityLoopSlot;
+        mLogger.info("Configuring Spark ID ", pSparkMax.getDeviceId(), " for velocity mode");
+    }
+
+
+    private void reloadVelocityGains(CANSparkMax pSparkMax) {
+        mLogger.info("Reloading gains for Talon ID ", pSparkMax.getDeviceId());
+
+        CANPIDController sparkMaxPid = pSparkMax.getPIDController();
+
+        sparkMaxPid.setSmartMotionAllowedClosedLoopError(SystemSettings.kDriveVelocityTolerance, SystemSettings.kDriveVelocityLoopSlot);
+        sparkMaxPid.setP(SystemSettings.kDriveVelocity_kP, SystemSettings.kDriveVelocityLoopSlot);
+        sparkMaxPid.setI(SystemSettings.kDriveVelocity_kI, SystemSettings.kDriveVelocityLoopSlot);
+        sparkMaxPid.setD(SystemSettings.kDriveVelocity_kD, SystemSettings.kDriveVelocityLoopSlot);
+        sparkMaxPid.setFF(SystemSettings.kDriveVelocity_kF, SystemSettings.kDriveVelocityLoopSlot);
+    }
+
+    private void configSparkForSmartMotion(CANSparkMax talon) {
+        configSparkForVelocity(talon);
+
+        talon.getPIDController().setSmartMotionMaxVelocity(SystemSettings.kDriveMotionMagicCruiseVelocity, SystemSettings.kLongCANTimeoutMs);
+        talon.getPIDController().setSmartMotionMaxAccel(SystemSettings.kDriveMotionMagicMaxAccel, SystemSettings.kLongCANTimeoutMs);
+    }
+
+    public Rotation2d getHeading() {
+        return mGyro.getHeading();
+    }
+
+    public double getLeftInches() {
```

Comment:
Since the conversion is specific to this drive train, does it make sense to do the conversion in a utility class?

Reply:
Yeah, we will need to refactor this eventually. Since `Conversions` is dependent on `SystemSettings` values, we should be able to get away with this short term by putting alternate values in `PracticeBotSystemSettings`